src/ranch_conns_sup_sup.erl

```diff
@@ -32,6 +32,8 @@ init({Ref, Transport, Protocol, Logger}) ->
 	TransOpts = ranch_server:get_transport_options(Ref),
 	NumAcceptors = maps:get(num_acceptors, TransOpts, 10),
 	NumConnsSups = maps:get(num_conns_sups, TransOpts, NumAcceptors),
+	StatsCounters = counters:new(2*NumConnsSups, []),
+	ok = ranch_server:set_stats_counters(Ref, StatsCounters),
```

Comment:
Come to think of it we should create this earlier probably? Maybe in `ranch_listener_sup`.

Reply:
I was undecided about this one. I went with the connections sup sup because it fetches the number of connection supervisors already. In other words, I was just being lazy ;)

Reply:
Ok, now I have a reason to keep it in ranch_conns_sup_sup...

The documentation (for ranch:set_transport_options) says that changes to num_conns_sups will take effect only when the entire listeners is restarted. This is not accurate (but good enough for the documentation, I guess): it will really take effect when _the conns_sup_sup_ restarts.

So, assuming the counters array is created in the listener_sup, when the number of conns_sups is being changed and afterwards _only the conns_sup_sup_ (ie, not the listener_sup) crashes and restarts, the number of counters in the array will not match the number of conns_sups any longer.

If OTOH the counters are created by ranch_conns_sup_sup, it will create a new counter array with the appropriate number of slots, matching the number of conns_sups it starts.

Reply:
OK.